### PR TITLE
Remove `ScalesListQuiet`

### PR DESCRIPTION
**Pull request by teunbrand**
TL;DR: This PR aims to fix compatibility with the new ggplot2 version.

Hi Malcolm,

We have been preparing a new release of ggplot2 and during a reverse dependency check, it became apparent that the prospective ggplot2 3.5.0 would break ggdag. Given that other packages also import ggdag, we felt it was prudent to help out to transition to the new version.

The culprit of the breakage was the `ScalesListQuiet` class, which (to my best estimate) re-implements `ggplot2:::ScalesList` with minor modifications. We made internal changes to `ggplot2:::ScalesList`, see https://github.com/tidyverse/ggplot2/pull/5144. However, as these changes aren't automatically transferred, ggdag would break.

This PR replaces the `ScalesListQuiet` class with a function that adds the custom `ScalesListQuiet$add()` method on the go. This should make ggdag immune to most\* changes in the `ggplot2:::ScalesList` class, and works both with the CRAN version and release candidate of ggplot2.

To test the code changes with the release candidate, you can install it with the code below:

```r
remotes::install_github("tidyverse/ggplot2", ref = remotes::github_pull("5592"))
```

The release of ggplot2 3.5.0 is scheduled at the end of January / early Februari. The progress can be tracked in https://github.com/tidyverse/ggplot2/issues/5588.
We are hoping that this PR might help ggside be ready around that time to soften the impact on downstream packages.

\* I've added a 'canary in the coal mine'-test for modifications to `ggplot2:::ScalesList$add()`.

### CODE REVIEW

**Comment by malcolmbarrett** [comment]
Thank you! I have a bigger release coming up but I'll try to make a patch release before then
